### PR TITLE
[Feature] Adds Expert Pokemon Breeder Mystery Encounter to the game

### PR DESCRIPTION
**Pull request by ben-lear**
## What are the changes the user will see?
Adds a new ME called "The Expert Pokemon Breeder"

See Issue: https://github.com/pagefaultgames/pokerogue/issues/3818

## Why am I making these changes?
Final Ultra ME that was being added for release

### Screenshots/Videos

https://github.com/user-attachments/assets/05b58d8c-a97f-4f73-970e-fd0447824eed

https://github.com/user-attachments/assets/267a2bf7-b17e-46c1-8102-819bf370f708

## How to test the changes?
can run locally and set `MYSTERY_ENCOUNTER_OVERRIDE` to `MysteryEncounterType.THE_EXPERT_BREEDER` (and set `MYSTERY_ENCOUNTER_RATE_OVERRIDE` to 256, and be on a valid ME wave)

## Checklist
- [ ] **I'm using `beta` as my base branch**
- [ ] There is no overlap with another PR?
- [ ] The PR is self-contained and cannot be split into smaller PRs?
- [ ] Have I provided a clear explanation of the changes?
- [ ] Have I considered writing automated tests for the issue?
- [ ] If I have text, did I make it translatable and add a key in the English locale file(s)?
- [ ] Have I tested the changes (manually)?
    - [ ] Are all unit tests still passing? (`npm run test`)
- [ ] Are the changes visual?
  - [ ] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Review by Adri1 — DISMISSED** [review]

File: src/test/mystery-encounter/encounters/the-expert-breeder-encounter.test.ts [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+
+  it("should not run below wave 10", async () => {
+    game.override.startingWave(9);
+
+    await game.runToMysteryEncounter();
+
+    expect(scene.currentBattle?.mysteryEncounter?.encounterType).not.toBe(MysteryEncounterType.THE_EXPERT_BREEDER);
+  });
+
+  it("should not run above wave 179", async () => {
```
Comment (DayKev) [inline src/test/mystery-encounter/encounters/the-expert-breeder-encounter.test.ts:96]: These 2 tests should not be in every ME test file, you're just duplicating the exact same test 50 times over. They should occur once, in some general ME test file.

**Review by CodeTappert — APPROVED** [review]

**Review by CodeTappert — APPROVED** [review]

**Review by damocleas — APPROVED** [review]

File: src/test/mystery-encounter/encounters/the-expert-breeder-encounter.test.ts [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+import { SelectModifierPhase } from "#app/phases/select-modifier-phase";
+import { TheExpertPokemonBreederEncounter } from "#app/data/mystery-encounters/encounters/the-expert-pokemon-breeder-encounter";
+import { TrainerType } from "#enums/trainer-type";
+import { EggTier } from "#enums/egg-type";
+import { PostMysteryEncounterPhase } from "#app/phases/mystery-encounter-phases";
+
+const namespace = "mysteryEncounter:expertPokemonBreeder";
+const defaultParty = [Species.LAPRAS, Species.GENGAR, Species.ABRA];
+const defaultBiome = Biome.CAVE;
+const defaultWave = 45;
```
Comment (DayKev) [inline src/test/mystery-encounter/encounters/the-expert-breeder-encounter.test.ts:25]: These should just be set directly in the `beforeEach`.

**Comment by DayKev** [comment]
```ts
const overrides = {
  MYSTERY_ENCOUNTER_RATE_OVERRIDE: 256,
  MYSTERY_ENCOUNTER_OVERRIDE: MysteryEncounterType.THE_EXPERT_BREEDER,
  STARTING_WAVE_OVERRIDE: 11,
  STARTING_LEVEL_OVERRIDE: 10
} satisfies Partial<InstanceType<typeof DefaultOverrides>>;
```
It didn't appear when I used these overrides, instead I got the Wobuffet ME (then after starting a new run I got some Oricorio ME).

**Review by Adri1 — APPROVED** [review]

**Review by DayKev — APPROVED** [review]

**Comment by chaosgrimmon** [comment]
What happens if the user's eggs are already full?

Also, seems like some text is cut off at the bottom: image

**Pull request merged**